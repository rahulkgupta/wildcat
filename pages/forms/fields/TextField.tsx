import { TextField } from '../../util/form';

interface Props {
    field: TextField
}


const TextFieldView = (props: Props) => {

    let field = props.field
    return (
        <div>
            <label>{field.getLabel()}</label>
            <input type="text" value={field.getValue()} onChange={(event) =>{field.setValue(event.target.value);}}></input>
        </div>
    )
}

export default TextFieldView;