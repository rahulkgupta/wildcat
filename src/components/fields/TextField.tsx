import TextField from '@src/util/form/fields/text';

interface Props {
  field: TextField;
}

/**
 * Renders the react view for a {@link TextField}
 * uses the various functions of {@link TextField} to handle display and updates.
 * @param props contains the {@link TextField}
 */
const TextFieldView = (props: Props) => {
  const { field } = props;
  return (
    <div>
      <label>{field.getLabel()}</label>
      <input
        type="text"
        value={field.getValue()}
        onChange={(event) => {
          field.setValue(event.target.value);
        }}
        data-testid={field.getId()}
      />
    </div>
  );
};

export default TextFieldView;
