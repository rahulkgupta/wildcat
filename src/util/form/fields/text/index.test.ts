import TextField from '.';

let textField: TextField;
let onUpdate: Function;

describe('TextField', () => {
  beforeEach(() => {
    onUpdate = jest.fn();
    textField = new TextField(
      '12234',
      {
        value: 'value',
        label: 'label',
        error: 'error',
      },
      onUpdate,
    );
  });

  test('getValue', () => {
    expect(textField.getValue()).toMatchSnapshot();
  });

  test('getId', () => {
    expect(textField.getId()).toMatchSnapshot();
  });

  test('getLabel', () => {
    expect(textField.getLabel()).toMatchSnapshot();
  });

  test('getError', () => {
    expect(textField.getError()).toMatchSnapshot();
  });

  test('setValue', () => {
    textField.setValue('34555');
    expect(textField.getValue()).toMatchSnapshot();
    expect(onUpdate).toHaveBeenCalled();
  });

  test('toJSON', () => {
    expect(textField.toJSON()).toMatchSnapshot();
  });
});
