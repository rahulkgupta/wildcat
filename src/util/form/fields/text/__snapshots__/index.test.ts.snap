// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField getError 1`] = `"error"`;

exports[`TextField getId 1`] = `"12234"`;

exports[`TextField getLabel 1`] = `"label"`;

exports[`TextField getValue 1`] = `"value"`;

exports[`TextField setValue 1`] = `"34555"`;

exports[`TextField toJSON 1`] = `
Object {
  "id": "12234",
  "value": "value",
}
`;
